### PR TITLE
Make corrections, rework sections for devcontainer.json reference

### PR DESCRIPTION
**Pull request by Chuxel**
This PR started out as some work to correct some devcontainer.json properties, add some key info, and add a number of missing ones.

However, as I was doing it, the tables started to get a bit out of control.

So, to fix that problem, I broke the properties into sections and did some de-duping to try to make it a bit easier to navigate.

### CODE REVIEW

**Comment by Chuxel** [comment]
Given the nature of this update - I'm assuming that this will likely float into next year for review, but wanted to get it up for feedback/thoughts.  //cc @2percentsilk as well

File: docs/remote/devcontainerjson-reference.md [hunk 1]
```diff
@@ -5,67 +5,103 @@ TOCTitle: devcontainer.json
 PageTitle: devcontainer.json reference
 ContentId: 52eaec33-21c6-410c-8e10-1ee3658a854f
 MetaDescription: devcontainer.json reference
-DateApproved: 12/8/2021
+DateApproved: 12/13/2021
 ---
 # devcontainer.json reference
 
-A `devcontainer.json` file in your project tells Visual Studio Code how to access (or create) a **development container** with a well-defined tool and runtime stack. This container can be used to run an application or to separate tools, libraries, or runtimes needed for working with a codebase. It can be used with the [Remote - Containers](https://aka.ms/vscode-remote/download/containers) extension or [GitHub Codespaces](https://github.com/features/codespaces).
+A `devcontainer.json` file in your project tells Visual Studio Code (and other services and tools that support the format) how to access (or create) a **development container** with a well-defined tool and runtime stack. It currently supported by the [Remote - Containers](https://aka.ms/vscode-remote/download/containers) extension and [GitHub Codespaces](https://github.com/features/codespaces).
```
Comment (bamurtaugh) [inline docs/remote/devcontainerjson-reference.md:12]:
```suggestion
A `devcontainer.json` file in your project tells Visual Studio Code (and other services and tools that support the format) how to access (or create) a **development container** with a well-defined tool and runtime stack. It's currently supported by the [Remote - Containers](https://aka.ms/vscode-remote/download/containers) extension and [GitHub Codespaces](https://github.com/features/codespaces).
```

File: docs/remote/devcontainerjson-reference.md [hunk 2]
```diff
@@ -5,67 +5,103 @@ TOCTitle: devcontainer.json
...
 
-[Set up a folder to run in a container](/docs/remote/create-dev-container.md#set-up-a-folder-to-run-in-a-container) has more information on configuring a dev container or use the **Remote-Containers: Add Development Container Configuration Files...** or **Codespaces: Add DevelopmentContainer Configuration Files...** commands from the Command Palette (`kbstyle(F1)`) to add a wide variety of base configurations from the [vscode-dev-containers repository](https://github.com/microsoft/vscode-dev-containers/tree/main/containers).
+[Set up a folder to run in a container](/docs/remote/create-dev-container.md#set-up-a-folder-to-run-in-a-container) has more information on configuring a dev container or you can use the **Remote-Containers: Add Development Container Configuration Files...** or **Codespaces: Add DevelopmentContainer Configuration Files...** commands from the Command Palette (`kbstyle(F1)`) to add a wide variety of base configurations from the [vscode-dev-containers repository](https://github.com/microsoft/vscode-dev-containers/tree/main/containers).
```
Comment (bamurtaugh) [inline docs/remote/devcontainerjson-reference.md:14]:
```suggestion
[Set up a folder to run in a container](/docs/remote/create-dev-container.md#set-up-a-folder-to-run-in-a-container) has more information on configuring a dev container or you can use the **Remote-Containers: Add Development Container Configuration Files...** or **Codespaces: Add Development Container Configuration Files...** commands from the Command Palette (`kbstyle(F1)`) to add a wide variety of base configurations from the [vscode-dev-containers repository](https://github.com/microsoft/vscode-dev-containers/tree/main/containers).
```

File: docs/remote/devcontainerjson-reference.md [hunk 3]
```diff
@@ -5,67 +5,103 @@ TOCTitle: devcontainer.json
...
 
 ## devcontainer.json properties
 
-While some devcontainer.json properties apply generally, others are only used in specific scenarios. The following table will outline the properties that apply in each situation.
+> **Tip:** If you've already built a container and connected to it, be sure to run **Remote-Containers: Rebuild Container** or **Codespaces: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up any changes you make change.
```
Comment (bamurtaugh) [inline docs/remote/devcontainerjson-reference.md:18]:
```suggestion
> **Tip:** If you've already built a container and connected to it, be sure to run **Remote-Containers: Rebuild Container** or **Codespaces: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up any changes you make.
```

**Review by bamurtaugh — COMMENTED** [review]
Thanks for making these changes!  Found a couple of minor tweaks, but I think it's looking great.

**Comment by Chuxel** [comment]
Thanks @bamurtaugh! Added the suggested changes.

**Pull request merged**